server/model/puzzle.ts

```diff
@@ -37,11 +38,18 @@ export async function listPuzzles(
       SELECT pid, uploaded_at, content
       FROM puzzles
       WHERE is_public = true
+      AND (content->'info'->>'type') = ANY($1)
+      AND ((content -> 'info' ->> 'title') || (content->'info'->>'author')) ILIKE ALL($2)
```

Comment:
This query here allows us to have the same searching logic as before (for each token is it in either the title or author?). It is a bit jank, but we can likely make an index to match this exact expression to make it quicker if desired.

Reply:
Note, I have not yet added indexes as I have not pulled in all of the puzzles locally, so this currently will make us do a full table scan.